tests/backend/specs/socketio.js

```diff
@@ -180,4 +186,54 @@ describe('socket.io access checks', () => {
     const message = await handshake(socket, 'other-pad');
     assert.equal(message.accessStatus, 'deny');
   });
+
+  // Authorization levels via authorize hook
+  it("level='create' -> can create", async () => {
+    authorize = () => 'create';
+    settings.requireAuthentication = true;
+    settings.requireAuthorization = true;
+    const res = await client.get('/p/pad').auth('user', 'user-password').expect(200);
+    socket = await connect(res);
+    const clientVars = await handshake(socket, 'pad');
+    assert.equal(clientVars.type, 'CLIENT_VARS');
+  });
+  it('level=true -> can create', async () => {
+    authorize = () => true;
+    settings.requireAuthentication = true;
+    settings.requireAuthorization = true;
+    const res = await client.get('/p/pad').auth('user', 'user-password').expect(200);
+    socket = await connect(res);
+    const clientVars = await handshake(socket, 'pad');
+    assert.equal(clientVars.type, 'CLIENT_VARS');
+  });
+  it("level='modify' -> can modify", async () => {
+    const pad = await padManager.getPad('pad'); // Create the pad.
+    authorize = () => 'modify';
+    settings.requireAuthentication = true;
+    settings.requireAuthorization = true;
+    const res = await client.get('/p/pad').auth('user', 'user-password').expect(200);
+    socket = await connect(res);
+    const clientVars = await handshake(socket, 'pad');
+    assert.equal(clientVars.type, 'CLIENT_VARS');
```

Comment:
Is this a thorough enough test for modify?  Just getting ``CLIENT_VARS`` seems a little trusting that ``CLIENT_VARS`` means you can modify??

Reply:
I added the following to the tests:

```
    assert.equal(clientVars.data.readonly, false);
```